### PR TITLE
fix: prevent Duplicate Block Explorer Entries and Ensure Proper Input Validation

### LINKED ISSUE #11990: [Bug]: Edit Network: Add Block Explorer duplicates existing block explorer without inputting value
### Describe the bug

When I edit a network, tap block explorer dropdown, tap on 'Add Block Explorer URL +' button and then tap 'Add Block Explorer URL' button without providing input to add another block explorer is added that duplicates the existing entry.

### Expected behavior

- 'Add Block Explorer URL' button should not be enabled until value is inputted into field.
- Block Explorer URL inputted should be validated after tapping 'Add Block Explorer URL' button
- Existing block explorer entries should not be re-added

### Screenshots/Recordings

https://github.com/user-attachments/assets/43019515-5a7a-4598-92d4-116610734b5b

### Steps to reproduce

GIVEN I am editing a network
AND I am adding a block explorer
WHEN I tap 'Add Block Explorer URL +'
THEN 'Add Block Explorer URL' button is disabled (FAILED: button is enabled)
AND 'Add Block Explorer URL' button remains disabled until value is 
WHEN I don't provide value for block explorer to add

### Error messages or log output

_No response_

### Detection stage

During release testing

### Version

7.34.0

### Build type

None

### Device

Any

### Operating system

iOS, Android

### Additional context

_No response_

### Severity

_No response_

### PR DESCRIPTION
**Pull request by salimtb**
## **Description**

When editing a network and attempting to add a new block explorer URL, the 'Add Block Explorer URL +' button allows users to add a duplicate entry even if no input is provided. This results in duplicate block explorer URLs being added to the list. The expected behavior is that the 'Add Block Explorer URL' button should remain disabled until valid input is provided, and existing URLs should not be re-added.

## **Related issues**

Fixes: #11990 

## **Manual testing steps**

1. Go to network add form
2. go to add block explorer url
3. The 'Add Block Explorer URL' button should be disabled until a valid URL is inputted into the field.

### **After**

https://github.com/user-attachments/assets/05f72bf8-8046-40a4-a941-51f39cfc9a09

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by salimtb** [comment]
The test coverage requirement has increased from 40% to a minimum of 60%. As a result, I've added additional test units unrelated to the fix, but necessary to ensure the build passes successfully.

File: app/components/Views/Settings/NetworksSettings/NetworkSettings/index.js [hunk 1]
```diff
@@ -2214,23 +2236,30 @@ export class NetworkSettings extends PureComponent {
...
-                  <View>
-                    <Text style={styles.warningText}>
-                      {strings('app_settings.invalid_block_explorer_url')}
-                    </Text>
-                  </View>
-                )}
+                {blockExplorerUrl &&
+                  (!isUrl(blockExplorerUrl) ||
+                    blockExplorerUrls.includes(blockExplorerUrlForm)) && (
+                    <View>
```
Comment (andreahaku) [inline app/components/Views/Settings/NetworksSettings/NetworkSettings/index.js:2242]: you don't need this `<view>` I think

**Review by andreahaku — COMMENTED** [review]
super small comment

File: app/components/Views/Settings/NetworksSettings/NetworkSettings/index.js (hunk 1 above)
Comment (salimtb) [reply app/components/Views/Settings/NetworksSettings/NetworkSettings/index.js]: fixed [7628802](https://github.com/MetaMask/metamask-mobile/pull/11995/commits/76288023569653ce6b19be22238df4232e6774b8)

**Review by andreahaku — APPROVED** [review]
LGTM

**Pull request merged**